File: python-gdb-activities-20/activity20/gdb/logging/log_destination.py
# gdb/logging/log_destination.py
from abc import ABC, abstractmethod
from typing import List
from gdb.domain.transaction import Transaction

class LogDestination(ABC):
    """Implementor Interface in the Bridge Pattern for log persistence destinations."""

    @abstractmethod
    def write(self, transaction: Transaction) -> None:
        """Write a transaction to the storage destination."""
        pass

    @abstractmethod
    def read_all(self) -> List[Transaction]:
        """Read all transactions from the storage destination."""
        pass

    @abstractmethod
    def get_destination_name(self) -> str:
        """Return human-readable destination identifier."""
        pass

    def readAll(self) -> List[Transaction]:
        return self.read_all()

    def getDestinationName(self) -> str:
        return self.get_destination_name()

File: java-gdb-activities-20/activity20-solution/src/com/gdb/logging/LogDestination.java
package com.gdb.logging;

import com.gdb.command.TransactionCommand;
import java.util.List;

/**
 * Implementor interface for Bridge Pattern decoupling transaction log storage backends.
 */
public interface LogDestination {
    void write(TransactionCommand cmd);
    List<TransactionCommand> readAll();
    void clear();
    String getDestinationName();
}
